### PR TITLE
[remote] wait until flush() is called to send output across the wire

### LINKED ISSUE #933: [remote] launcher CommandSession/GogoRedirector output can be slow
When using -runremote and specifying the shell to use "CommandSession/GogoRedirector" (specify the shell=-1) if I send a command in the eclipse console like "lb" it will return the output line by line, where it takes maybe 15-20 seconds to output a standard "lb" command with maybe 100 bundles.

### PR DESCRIPTION
**Pull request by gamerson**
Fixes #933

### CODE REVIEW

File: biz.aQute.remote/src/aQute/remote/agent/Shell.java [hunk 1]
```diff
@@ -49,6 +49,7 @@ public synchronized void add(String s) throws IOException {
 			running = true;
 			try {
 				Object result = session.execute(s);
+				this.out.flush();
```
Comment (bjhargrave) [inline biz.aQute.remote/src/aQute/remote/agent/Shell.java]: What about flush after the println and printStackTrace below? Perhaps the flush call should be at the bottom of the loop before the call to prompt?

File: biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java [hunk 2]
```diff
@@ -85,6 +69,31 @@ public void write(byte b[], int off, int len) {
...
+				try {
+					agent.close();
+				}
+				catch (IOException e) {
+					// e.printStackTrace();
+				}
+			}
+		}
+
+		sb = new StringBuilder();
```
Comment (bjhargrave) [inline biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java]: Perhaps you should convert the StringBuffer to a local String (and use in the loop) and reset sb before the for loop?

File: biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java [hunk 3]
```diff
@@ -85,6 +69,31 @@ public void write(byte b[], int off, int len) {
 
 	public void flush() {
 		super.flush();
```
Comment (bjhargrave) [inline biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java]: I am not sure of the proper order here. Should the new code be executed before super.flush()?

Comment (gamerson) [inline biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java]: now that you mention it, I think before super.flush() makes more sense.

File: biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java (hunk 2 above)
Comment (gamerson) [inline biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java]: Done

File: biz.aQute.remote/src/aQute/remote/agent/Shell.java (hunk 1 above)
Comment (gamerson) [inline biz.aQute.remote/src/aQute/remote/agent/Shell.java]: agree

**Pull request merged**